### PR TITLE
Refactor handlers

### PR DESCRIPTION
**Pull request by thani-sh**

### CODE REVIEW

**Comment by arunoda** [comment]
Machan, how about this one. Then testing and everything will be simple.

``` go
type MessageHandle func(*Server, Response, Message) error

// These are the handles

func connectHandle(s *Server, res Response, m Message) error {
 return res.WriteJSON(map[string]string{
   "msg":     "connected",
   "session": random.Id(17),
 })
}

func pingHandle(s *Server, res Response, m Message) error {
 ...
}

func methodHandle(s *Server, res Response, m Message) error {
 ...
}

DDPHandlers:= map[string]MessageHandle {
  "connect": connectHandle,
  "ping": pingHandle,
  "method": methodHandle
}

// In the server.

if handler, ok := DDPHandlers[msg.Msg]; ok {
  go handler(s, res, msg)
} else {
  break
}
```

**Comment by thani-sh** [comment]
Yeah. Code is a lot simpler this way.

**Comment by arunoda** [comment]
great.

**Comment by thani-sh** [comment]
Added some integration tests

File: server/integration/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	if err != nil {
+		t.Error("websocket connection failed")
+	}
+
+	defer ws.Close()
+
+	writeMessage(ws, `{"msg": "connect", "version": "1", "support": ["1"]}`, t)
+	msg := readMessage(ws, t)
+
+	if msg.Msg != "connected" {
```
Comment (arunoda) [inline server/integration/integration_test.go:66]: duplicate.

**Pull request merged**